libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts

```diff
@@ -1,118 +1,76 @@
+  // cacheController is the controller used to cache the evaluation of the flags.
+  private readonly _cacheController?: CacheController;
+  private _pollingIntervalId?: number;
+  private _pollingInterval: number;
 
-  readonly hooks?: Hook[];
   constructor(options: GoFeatureFlagProviderOptions, logger?: Logger) {
-    this.timeout = options.timeout || 0; // default is 0 = no timeout
-    this.endpoint = options.endpoint;
-
-    if (!options.disableDataCollection) {
-      this.dataCollectorHook = new GoFeatureFlagDataCollectorHook(
-        {
-          endpoint: this.endpoint,
-          timeout: this.timeout,
-          dataFlushInterval: options.dataFlushInterval,
-        },
-        logger,
-      );
-      this.hooks = [this.dataCollectorHook];
-    }
-
-    this.cacheTTL = options.flagCacheTTL !== undefined && options.flagCacheTTL !== 0 ? options.flagCacheTTL : 1000 * 60;
-
-    this.disableDataCollection = options.disableDataCollection || false;
-    this.logger = logger;
-
-    // Add API key to the headers
-    if (options.apiKey) {
-      axios.defaults.headers.common['Authorization'] = `Bearer ${options.apiKey}`;
-    }
-
-    if (!options.disableCache) {
-      const cacheSize =
-        options.flagCacheSize !== undefined && options.flagCacheSize !== 0 ? options.flagCacheSize : 10000;
-      this.cache = new LRUCache({ maxSize: cacheSize, sizeCalculation: () => 1 });
-    }
+    this._goffApiController = new GoffApiController(options);
+    this._dataCollectorHook = new GoFeatureFlagDataCollectorHook(
+      { dataFlushInterval: options.dataFlushInterval },
+      this._goffApiController,
+      logger,
+    );
+    this._disableDataCollection = options.disableDataCollection || false;
+    this._cacheController = new CacheController(options, logger);
+    this._pollingInterval = options.pollInterval ?? this.DEFAULT_POLL_INTERVAL;
   }
 
   /**
    * initialize is called everytime the provider is instanced inside GO Feature Flag.
    * It will start the background process for data collection to be able to run every X ms.
    */
   async initialize() {
-    if (!this.disableDataCollection) {
-      this.dataCollectorHook?.init();
+    if (!this._disableDataCollection && this._dataCollectorHook) {
+      this.hooks = [this._dataCollectorHook];
+      this._dataCollectorHook.init();
     }
-    this._status = ProviderStatus.READY;
-  }
 
-  get status() {
-    return this._status;
+    if (this._pollingInterval > 0) {
+      this.startPolling();
+    }
   }
 
   /**
    * onClose is called everytime OpenFeature.Close() function is called.
-   * It will terminate gracefully the provider and ensure that all the data are send to the relay-proxy.
+   * It will gracefully terminate the provider and ensure that all the data are sent to the relay-proxy.
    */
   async onClose() {
-    await this.dataCollectorHook?.close();
+    this._cacheController?.clear();
+    await this._dataCollectorHook?.close();
+    this.stopPolling();
```

Comment:
This is also a bit and I guess not important but we should first stop polling and clear the cache then.
```suggestion
    this.stopPolling();
    this._cacheController?.clear();
    await this._dataCollectorHook?.close();
```